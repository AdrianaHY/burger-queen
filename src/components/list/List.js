import React, { Component } from 'react';

import ListItem from './ListItem';

import './List.scss';

class List extends Component {
	render() {
		const { breakfastMenu, breakfast, dinnerAndLunchMenu, addItem, total } = this.props;
		return (
			<div className="list-container">
				<div className="list">
				{
					(breakfast ? breakfastMenu : dinnerAndLunchMenu).map((item) => (
						<ListItem
							item={item}
							key={item.name}
							addItem={addItem}
							breakfast={breakfast}
						/>
					))
				}
				</div>
				<div className="total">
					<p>
						<span>Total</span>{' '}
						<span>{`$${total}.00`}</span>
					</p>
				</div>
			</div>
		)
	}
}

export default List;