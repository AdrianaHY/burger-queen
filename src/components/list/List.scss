@import '../../styles/_variables.scss';

.list-container {
	display: flex;
	flex-direction: row;
	justify-content: space-between;

	.list {
		width: 50%
	}

	.total {
		display: flex;
		align-items: center;
		justify-content: center;
		width: 50%;
		font-family: $fontFamily;
		font-size: $fontSizeHuge;
		padding: $paddingMedium;

		p {
			background: $lightYellow;
			padding: $paddingLarge $paddingHuge;
		}
	}
}