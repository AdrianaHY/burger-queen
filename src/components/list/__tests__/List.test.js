import React from 'react';
import { shallow } from 'enzyme';
import List from '../List';

describe('components | list | List', () => {
	it('Renders the component' , () => {
		const props = {
			breakfast: true,
			breakfastMenu: [
				'Sandwich',
				'Coffe'
			],
			dinnerAndLunchMenu: [
				'Simple Burger',
				'Large Soda'
			],
			addItem: jest.fn(),
			total: 50
		};

		const component = shallow(
			<List {...props} />
		);

		expect(component).toMatchSnapshot();
	})
})