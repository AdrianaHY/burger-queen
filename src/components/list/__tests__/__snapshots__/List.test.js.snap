// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | list | List Renders the component 1`] = `ShallowWrapper {}`;
